app/components/home/GuestHome/Dialog/signUp/SignUpForm.jsx

```diff
@@ -0,0 +1,220 @@
+    this.setState({ disableButton: !ready });
+  }
+
+  isAlphanumeric(value) {
+    return /^[a-zA-Z0-9_]/.test(value) ?
+      '' :
+      'This value must be alphanumeric';
+  }
+
+  isEmail(value) {
+    return /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(value) ?
+      '' :
+      'A valid email is required';
+  }
+
+  validate(field) {
+    switch (field) {
+      case ('username'):
+        return this.state.username ?
+          this.isAlphanumeric(this.state.username) :
+          'This field is required';
+      case ('firstName'):
+        return this.state.firstName ?
+          this.isAlphanumeric(this.state.firstName) :
+          'This field is required';
+      case ('lastName'):
+        return this.state.lastName ?
+          this.isAlphanumeric(this.state.lastName) :
+          'This field is required';
+      case ('email'):
+        return this.state.email ?
+          this.isEmail(this.state.email) :
+          'This field is required';
+      case ('password'):
+        return this.state.password ?
+          '' :
+          'This field is required';
+      default:
+        return '';
+    }
+  }
+  signUpUser() {
+    request
+      .post('api/users')
+      .send({
+        username: this.state.username,
+        firstName: this.state.firstName,
+        lastName: this.state.lastName,
+        email: this.state.email,
+        password: this.state.password
+      })
+      .end((error, response) => {
+        if (response.statusCode === 409) {
+          return this.setState({
+            snackBarMessage: 'Username or email already exists',
+            snackBarOpen: true,
+            username: '',
+            email: ''
+          });
+        }
+
+        return this.setState({
+          snackBarMessage: 'Account created, Log in to continue.',
+          snackBarOpen: true,
+          username: '',
+          firstName: '',
+          lastName: '',
+          email: '',
+          password: ''
+        });
+      });
+  }
+
+  render() {
+    return (
+      <div style={divStyle}>
+        <div style={fieldStyle}>
+          <TextField
+            style={{ width: '100%' }}
+            hintText="Username"
+            floatingLabelText="Username"
+            value={this.state.username}
+            errorText={this.validate('username')}
+            onBlur={this.formReady}
+            onChange={
+              (event) => {
+                this.setState({ username: event.target.value });
+              }
+            }
+          />
+        </div>
+        <div style={fieldStyle}>
+          <TextField
+            style={{ width: '49%', float: 'left' }}
+            hintText="First Name"
+            floatingLabelText="First Name"
+            errorText={this.validate('firstName')}
+            value={this.state.firstName}
+            onBlur={this.formReady}
+            onChange={
+              (event) => {
+                this.setState({ firstName: event.target.value });
```

Comment:
**Code Quality - Nitpick**
You should consider abstracting this into a generic function, f.e.
you could add a name field to every text field, then have a function that you use as the `onchange` handler that may look like
```javascript
...
handleFieldChange(event) {
  this.setState(Object.assign({}, this.state, {
    [event.target.name]: event.target.value.trim()
  }
}
...
```
Note that I haven't tried out the above snippet, but you get the idea. 😄